lib/faraday/connection.rb

```diff
@@ -163,8 +163,16 @@ def headers=(hash)
     METHODS_WITH_QUERY.each do |method|
       class_eval <<-RUBY, __FILE__, __LINE__ + 1
         def #{method}(url = nil, params = nil, headers = nil)
```

Comment:
I'm sorry my example in that comment wasn't probably the best one as it ignored the `headers` param.

I think it would be much easier if you use Ruby keywords arguments (a.k.a. named params).
The signature of this method would become `def #{method}(url = nil, params = nil, headers = nil, body:)` and you could just refer to `body` in the method implementation without checking for particular keys in headers.

This way you can also specify headers AND a body: `get('/path', params, headers, body: { ... })`

Reply:
Ah I see, Ruby still allows you to specify a hash param that way 🤦‍♂️.